### PR TITLE
Fix #3015 - fix RtlGenRandom function signature

### LINKED ISSUE #3015: RtlGenRandom definition is incorrect
The definition for `RtlGenRandom` in `std/os/windows/advapi32.zig` is incorrect:

```
pub extern "advapi32" stdcallcc fn SystemFunction036(output: [*]u8, length: usize) BOOL;
pub const RtlGenRandom = SystemFunction036;
```

The actual definition is:

```
BOOLEAN RtlGenRandom(
  PVOID RandomBuffer,
  ULONG RandomBufferLength
);
```

Where `ULONG` is a 32 bit unsigned number.

Note that this will also require a change in `std/os/windows.zig` to handle calls to this function as `usize` is 64 bits on 64 bit systems.

### PR DESCRIPTION
**Pull request by euantorano**
Fixes #3015

### CODE REVIEW

File: std/os/windows.zig [hunk 1]
```diff
@@ -138,10 +138,19 @@ pub const RtlGenRandomError = error{Unexpected};
...
 pub fn RtlGenRandom(output: []u8) RtlGenRandomError!void {
-    if (advapi32.RtlGenRandom(output.ptr, output.len) == 0) {
-        switch (kernel32.GetLastError()) {
-            else => |err| return unexpectedError(err),
+    var total_read: usize = 0;
+    var buff: []u8 = output[0..];
+    const max_read_size: ULONG = ULONG(maxInt(ULONG));
+
+    while (total_read < output.len) {
+        const to_read: ULONG = @intCast(ULONG, math.min(buff.len, max_read_size));
```
Comment (andrewrk) [inline std/os/windows.zig]: I've pushed a commit to master (2cd5e555818583e77e5601d43d55339e8c4017b0) that makes this `@intCast` unnecessary :-)

File: std/os/windows.zig [hunk 2]
```diff
@@ -138,10 +138,19 @@ pub const RtlGenRandomError = error{Unexpected};
 /// https://github.com/rust-lang-nursery/rand/issues/111
 /// https://bugzilla.mozilla.org/show_bug.cgi?id=504270
 pub fn RtlGenRandom(output: []u8) RtlGenRandomError!void {
-    if (advapi32.RtlGenRandom(output.ptr, output.len) == 0) {
-        switch (kernel32.GetLastError()) {
-            else => |err| return unexpectedError(err),
+    var total_read: usize = 0;
+    var buff: []u8 = output[0..];
+    const max_read_size: ULONG = ULONG(maxInt(ULONG));
```
Comment (andrewrk) [inline std/os/windows.zig]: One of those ULONG's is redundant:

```suggestion
    const max_read_size: ULONG = maxInt(ULONG);
```

File: std/os/windows.zig [hunk 3]
```diff
@@ -138,10 +138,19 @@ pub const RtlGenRandomError = error{Unexpected};
...
+
+    while (total_read < output.len) {
+        const to_read: ULONG = @intCast(ULONG, math.min(buff.len, max_read_size));
+
+        if (advapi32.RtlGenRandom(buff.ptr, to_read) == 0) {
+            return unexpectedError(kernel32.GetLastError());
         }
+
+        total_read += @intCast(usize, to_read);
```
Comment (andrewrk) [inline std/os/windows.zig]: I believe ULONG is guaranteed to be always >= the bits of usize, and so this `@intCast` is not needed.

Comment (euantorano) [reply std/os/windows.zig]: Yep, it's a u32 so it's always <= the size of `usize` (I believe that's what you meant 😉)

**Pull request merged**